File: android/services-v72/src/main/java/com/liferay/mobile/android/v72/segmentsexperience/SegmentsexperienceService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v72.segmentsexperience;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class SegmentsexperienceService extends BaseService {

	public SegmentsexperienceService(Session session) {
		super(session);
	}

	public JSONObject addSegmentsExperience(long segmentsEntryId, long classNameId, long classPK, JSONObject nameMap, boolean active, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("classNameId", classNameId);
			_params.put("classPK", classPK);
			_params.put("nameMap", checkNull(nameMap));
			_params.put("segmentsEntryId", segmentsEntryId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/segments.segmentsexperience/add-segments-experience", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject deleteSegmentsExperience(long segmentsExperienceId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("segmentsExperienceId", segmentsExperienceId);

			_command.put("/segments.segmentsexperience/delete-segments-experience", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getSegmentsExperience(long segmentsExperienceId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("segmentsExperienceId", segmentsExperienceId);

			_command.put("/segments.segmentsexperience/get-segments-experience", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getSegmentsExperiences(long groupId, long classNameId, long classPK, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("classNameId", classNameId);
			_params.put("classPK", classPK);
			_params.put("groupId", groupId);

			_command.put("/segments.segmentsexperience/get-segments-experiences", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getSegmentsExperiences(long groupId, long classNameId, long classPK, boolean active, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("classNameId", classNameId);
			_params.put("classPK", classPK);
			_params.put("end", end);
			_params.put("groupId", groupId);
			_params.put("start", start);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.segments.model.SegmentsExperience>", orderByComparator);

			_command.put("/segments.segmentsexperience/get-segments-experiences", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getSegmentsExperiencesCount(long groupId, long classNameId, long classPK, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("classNameId", classNameId);
			_params.put("classPK", classPK);
			_params.put("groupId", groupId);

			_command.put("/segments.segmentsexperience/get-segments-experiences-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject updateSegmentsExperience(long segmentsExperienceId, long segmentsEntryId, JSONObject nameMap, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("nameMap", checkNull(nameMap));
			_params.put("segmentsEntryId", segmentsEntryId);
			_params.put("segmentsExperienceId", segmentsExperienceId);

			_command.put("/segments.segmentsexperience/update-segments-experience", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void updateSegmentsExperiencePriority(long segmentsExperienceId, int newPriority) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("newPriority", newPriority);
			_params.put("segmentsExperienceId", segmentsExperienceId);

			_command.put("/segments.segmentsexperience/update-segments-experience-priority", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

}

File: windows/Liferay.SDK/Service/V73/Segmentsexperience/SegmentsexperienceService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Segmentsexperience
{
	public class SegmentsexperienceService : ServiceBase
	{
		public SegmentsexperienceService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> AddSegmentsExperienceAsync(long segmentsEntryId, long classNameId, long classPK, IDictionary<string, string> nameMap, bool active, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("segmentsEntryId", segmentsEntryId);
			_parameters.Add("classNameId", classNameId);
			_parameters.Add("classPK", classPK);
			_parameters.Add("nameMap", nameMap);
			_parameters.Add("active", active);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperience/add-segments-experience", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateSegmentsExperienceAsync(long segmentsExperienceId, long segmentsEntryId, IDictionary<string, string> nameMap, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("segmentsExperienceId", segmentsExperienceId);
			_parameters.Add("segmentsEntryId", segmentsEntryId);
			_parameters.Add("nameMap", nameMap);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperience/update-segments-experience", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetSegmentsExperienceAsync(long segmentsExperienceId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("segmentsExperienceId", segmentsExperienceId);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperience/get-segments-experience", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> FetchSegmentsExperienceAsync(long groupId, string segmentsExperienceKey)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("segmentsExperienceKey", segmentsExperienceKey);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperience/fetch-segments-experience", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetSegmentsExperiencesAsync(long groupId, long classNameId, long classPK, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("classNameId", classNameId);
			_parameters.Add("classPK", classPK);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperience/get-segments-experiences", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetSegmentsExperiencesAsync(long groupId, long classNameId, long classPK, bool active, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("classNameId", classNameId);
			_parameters.Add("classPK", classPK);
			_parameters.Add("active", active);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperience/get-segments-experiences", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<dynamic> DeleteSegmentsExperienceAsync(long segmentsExperienceId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("segmentsExperienceId", segmentsExperienceId);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperience/delete-segments-experience", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<long> GetSegmentsExperiencesCountAsync(long groupId, long classNameId, long classPK, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("classNameId", classNameId);
			_parameters.Add("classPK", classPK);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperience/get-segments-experiences-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task UpdateSegmentsExperiencePriorityAsync(long segmentsExperienceId, int newPriority)
		{
			var _parameters = new JsonObject();

			_parameters.Add("segmentsExperienceId", segmentsExperienceId);
			_parameters.Add("newPriority", newPriority);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperience/update-segments-experience-priority", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}
	}
}